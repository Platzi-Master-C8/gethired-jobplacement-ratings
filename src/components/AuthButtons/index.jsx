import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
