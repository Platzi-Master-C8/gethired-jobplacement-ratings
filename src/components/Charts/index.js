import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }