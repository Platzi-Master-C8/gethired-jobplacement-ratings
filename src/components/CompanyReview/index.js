import CompanyReview from './CompanyReview';

export { CompanyReview };
