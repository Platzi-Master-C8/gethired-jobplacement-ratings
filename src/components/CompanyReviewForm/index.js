import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
