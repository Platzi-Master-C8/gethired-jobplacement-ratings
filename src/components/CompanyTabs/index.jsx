import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
