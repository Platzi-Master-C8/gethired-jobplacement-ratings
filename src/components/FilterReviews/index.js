import FilterReviews from './FilterReviews';

export {
    FilterReviews
}