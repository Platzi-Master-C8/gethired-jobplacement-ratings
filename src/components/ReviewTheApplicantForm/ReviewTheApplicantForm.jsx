import React, { useState } from 'react';

import {
    FormControl,
    OutlinedInput,
    Grid,
    Typography,
    FormControlLabel,
    Box,
    Button,
    Modal,
    RadioGroup,
    Radio,
} from '@mui/material';

import { styled } from '@mui/material/styles';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import StarIcon from '@mui/icons-material/Star';

const CustomBox = styled(Box)({
    display: 'flex',

    '& .MuiSvgIcon-root': {
        fontSize: 30,
        cursor: 'pointer',
        margin: '0 10px',
    },
});

const boxStyles = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: '45%',
    height: '75%',
    bgcolor: 'background.paper',
    boxShadow: 24,
    p: 4,
    overflow: 'hidden',
};

const ReviewTheApplicantForm = () => {
    const [open, setOpen] = useState(false);

    const [review, setReview] = useState({
        applicant_name: '',
        is_hired: null,
        communication_rating: undefined,
        confidence_rating: undefined,
        negotiation_rating: undefined,
        motivation_rating: undefined,
        self_knowledge_rating: undefined,
        hard_skill_rating: undefined,
    });

    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);

    const handleInput = (e) => {
        const { name, value } = e.target;
        setReview({ ...review, [name]: value });
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(review);
        setOpen(false);
    };

    return (
        <div>
            <Grid md={12} sx={{ display: 'grid', justifyContent: 'flex-end' }}>
                <Button onClick={handleOpen}>Review the applicant </Button>
            </Grid>
            <Modal
                open={open}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <form onSubmit={handleSubmit}>
                    <Box sx={boxStyles}>
                        <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                            Review the Applicant
                        </Typography>
                        <Grid container marginTop={5}>
                            <Grid item md={6}>
                                <Typography variant="subtitle1" marginBottom="20px" sx={{ fontSize: 18 }}>
                                    Applicant name
                                </Typography>
                                <FormControl sx={{ width: '85%' }} required>
                                    <OutlinedInput
                                        id="applicant_name"
                                        name="applicant_name"
                                        placeholder="Aplicant name"
                                        value={review.applicant_name}
                                        onChange={handleInput}
                                    />
                                </FormControl>
                            </Grid>
                            <Grid item md={6}>
                                <Typography variant="subtitle1" marginBottom="30px" sx={{ fontSize: 18 }}>
                                    Was he/she hired?
                                </Typography>
                                <FormControl component="fieldset" required>
                                    <RadioGroup
                                        id="is_hired"
                                        name="is_hired"
                                        row
                                        value={review.is_hired}
                                        onChange={handleInput}
                                    >
                                        <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                        <FormControlLabel value={0} control={<Radio />} label="No" />
                                    </RadioGroup>
                                </FormControl>
                            </Grid>
                        </Grid>
                        <Grid marginTop={3}>
                            <Typography variant="h2">Skill Assessment</Typography>
                            <Grid container rowSpacing={1} alignItems="center" marginTop="5px">
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Communication</Typography>
                                </Grid>
                                <Grid item md={6} sm={12}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.communication_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, communication_rating: index + 1 })
                                                        }
                                                        sx={{ fontSize: 30 }}
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, communication_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Confidence</Typography>
                                </Grid>
                                <Grid item md={6} sm={12}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.confidence_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, confidence_rating: index + 1 })
                                                        }
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, confidence_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Negotiation</Typography>
                                </Grid>
                                <Grid item md={6}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.negotiation_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, negotiation_rating: index + 1 })
                                                        }
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, negotiation_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Motivation</Typography>
                                </Grid>
                                <Grid item md={6} sm={12}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.motivation_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, motivation_rating: index + 1 })
                                                        }
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, motivation_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Self-knowledge</Typography>
                                </Grid>
                                <Grid item md={6} sm={12}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.self_knowledge_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, self_knowledge_rating: index + 1 })
                                                        }
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, self_knowledge_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                                <Grid item md={5} sm={12}>
                                    <Typography variant="subtitle2">Hard Skills</Typography>
                                </Grid>
                                <Grid item md={6} sm={12}>
                                    <CustomBox>
                                        {[...Array(5)].map((star, index) => (
                                            <div>
                                                {index < review.hard_skill_rating ? (
                                                    <StarIcon
                                                        onClick={() =>
                                                            setReview({ ...review, hard_skill_rating: index + 1 })
                                                        }
                                                    />
                                                ) : (
                                                    <StarBorderIcon
                                                        onClick={() =>
                                                            setReview({ ...review, hard_skill_rating: index + 1 })
                                                        }
                                                    />
                                                )}
                                            </div>
                                        ))}
                                    </CustomBox>
                                </Grid>
                            </Grid>
                        </Grid>
                        <Grid container sx={{ alignItems: 'center', justifyContent: 'center', marginTop: 5 }}>
                            <Grid item md={2} onClick={handleClose}>
                                <Button variant="outlined">Cancel</Button>
                            </Grid>
                            <Grid item md={2}>
                                <Button variant="outlined" type="submit">
                                    Submit
                                </Button>
                            </Grid>
                        </Grid>
                    </Box>
                </form>
            </Modal>
        </div>
    );
};

export default ReviewTheApplicantForm;
