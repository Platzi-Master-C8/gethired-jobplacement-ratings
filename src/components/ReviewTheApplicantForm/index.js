import ReviewTheApplicantForm from "./ReviewTheApplicantForm";

export { ReviewTheApplicantForm };