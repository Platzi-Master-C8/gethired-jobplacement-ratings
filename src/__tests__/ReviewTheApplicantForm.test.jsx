import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ReviewTheApplicantForm } from '../components/ReviewTheApplicantForm';
import MediaQueyProvider from '../context/MediaQueryContext';

const renderComponent = () =>
    render(
        <MediaQueyProvider>
            <ReviewTheApplicantForm />
        </MediaQueyProvider>,
    );

describe('Component <ReviewTheApplicantForm/>', () => {
    beforeEach(() => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () =>
                    Promise.resolve({
                        applicant_name: 'Test Applicant',
                        is_hired: '1',
                        communication_rating: 4,
                        confidence_rating: 4,
                        negotiation_rating: 4,
                        motivation_rating: 4,
                        self_knowledge_rating: 4,
                        hard_skill_rating: 4,
                    }),
            }),
        );
    });

    test('renders ReviewTheApplicantForm', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    test('the modal opens correctly', () => {
        const subject = renderComponent();

        fireEvent.click(subject.getByText('Review the applicant'));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();
    });

    test('it does close the Modal', () => {
        const subject = renderComponent();

        fireEvent.click(subject.getByText('Review the applicant'));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();

        fireEvent.click(subject.getByText(/Cancel/));
        expect(subject.container.querySelectorAll('button').length).toBe(1);
    });

    test('it does not send when try to send without fill anything', () => {
        const subject = renderComponent();

        fireEvent.click(subject.getByText('Review the applicant'));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();

        fireEvent.click(subject.getByText(/Submit/));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();
    });

    test('it allows to fill the inputs', () => {
        const subject = renderComponent();

        fireEvent.click(subject.getByText('Review the applicant'));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();

        const applicantName = subject.getByPlaceholderText('Applicant name');
        fireEvent.change(applicantName, { target: { value: 'Developer' } });
        expect(applicantName.value).toBe('Developer');
    });

    test('it shows and error mensagge when the raitings are not selected', () => {
        const subject = renderComponent();

        fireEvent.click(subject.getByText('Review the applicant'));
        expect(subject.getByText('Applicant name')).toBeInTheDocument();

        const applicantName = subject.getByPlaceholderText('Applicant name');
        fireEvent.change(applicantName, { target: { value: 'Developer' } });
        expect(applicantName.value).toBe('Developer');

        fireEvent.click(subject.getByText(/Submit/));
        const alertBar = subject.getByRole('alert');
        expect(alertBar).toBeInTheDocument();

        const isHired = subject.getByTestId('is_hired');
        fireEvent.click(isHired.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar1 = subject.getByRole('alert');
        expect(alertBar1).toBeInTheDocument();

        const communicationR = subject.getByTestId('communication_rating');
        fireEvent.click(communicationR.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar2 = subject.getByRole('alert');
        expect(alertBar2).toBeInTheDocument();

        const confidenceR = subject.getByTestId('confidence_rating');
        fireEvent.click(confidenceR.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar3 = subject.getByRole('alert');
        expect(alertBar3).toBeInTheDocument();

        const negotiationR = subject.getByTestId('negotiation_rating');
        fireEvent.click(negotiationR.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar4 = subject.getByRole('alert');
        expect(alertBar4).toBeInTheDocument();

        const motivationR = subject.getByTestId('motivation_rating');
        fireEvent.click(motivationR.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar5 = subject.getByRole('alert');
        expect(alertBar5).toBeInTheDocument();

        const selfKnowledge = subject.getByTestId('self_knowledge_rating');
        fireEvent.click(selfKnowledge.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        const alertBar6 = subject.getByRole('alert');
        expect(alertBar6).toBeInTheDocument();

        const hardSkills = subject.getByTestId('hard_skill_rating');
        fireEvent.click(hardSkills.querySelectorAll('input')[0]);
        fireEvent.click(subject.getByText(/Submit/));
        expect(subject.container.querySelectorAll('button').length).toBe(1);
    });
});
