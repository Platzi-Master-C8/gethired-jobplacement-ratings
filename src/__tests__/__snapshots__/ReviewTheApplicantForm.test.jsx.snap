// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ReviewTheApplicantForm/> renders ReviewTheApplicantForm 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-oav0mg-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Review the applicant 
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
